openslides_backend/action/action_worker.py

```diff
@@ -0,0 +1,288 @@
+                    "name": action_worker_writing.action_names,
+                    "written": action_worker_writing.written,
+                }
+            ]
+        ],
+    )
+
+
+class ActionWorkerWriting(object):
+    def __init__(
+        self,
+        user_id: int,
+        starttime: int,
+        logging: LoggingModule,
+        action_names: str,
+        datastore: DatastoreService,
+    ) -> None:
+        self.user_id = user_id
+        self.starttime = starttime
+        self.logger = logging.getLogger(__name__)
+        self.action_names = action_names
+        self.datastore = datastore
+
+        self.new_id: Optional[int] = None
+        self.fqid: str = "Still not set"
+        self.written: bool = False
+
+    def initial_action_worker_write(self) -> str:
+        current_time = round(time())
+        with self.datastore.get_database_context():
+            if not self.new_id:
+                self.new_id = self.datastore.reserve_id(collection="action_worker")
+                self.fqid = fqid_from_collection_and_id("action_worker", self.new_id)
+            try:
+                self.datastore.write_action_worker(
+                    WriteRequest(
+                        events=[
+                            Event(
+                                type=EventType.Create,
+                                fqid=self.fqid,
+                                fields={
+                                    "id": self.new_id,
+                                    "name": self.action_names,
+                                    "state": "running",
+                                    "created": self.starttime,
+                                    "timestamp": current_time,
+                                },
+                            )
+                        ],
+                        information={self.fqid: ["create action_worker"]},
+                        user_id=self.user_id,
+                        locked_fields={},
+                    )
+                )
+                self.datastore.get(
+                    self.fqid, [], lock_result=False, use_changed_models=False
+                )
+                message = f"Action lasts too long. {self.fqid} written to database. Get the result from database, when the job is done."
+                self.written = True
+            except DatastoreException as e:
+                message = f"Action lasts too long, exception on writing {self.fqid}: {e.message}. Get the result later from database."
+        self.logger.debug(f"action_worker: {message}, written:{self.written}")
+        return message
+
+    def continue_action_worker_write(self) -> None:
+        current_time = round(time())
+        with self.datastore.get_database_context():
+            ram = psutil.virtual_memory()
+            response = {
+                "success": False,
+                "message": f"ram total:{ram.total} available:{ram.available} percent:{ram.percent} used:{ram.used} free.{ram.free}",
```

Comment:
Do you really want to spam the DB with these informations? I would think they rather belong in the logs instead of the db, and maybe not on each iteration, but on each xth (x=100? 1000?).

Reply:
This is the "Still alive"-signal for the client or anybody else, who wants to wait for the end of the action. It will be written every 10seconds, see lock.acquire in method gunicorn_post_request.
I shortened the memory information, but for now, it may be usefull.